portal-impl/src/com/liferay/portlet/asset/service/impl/AssetCategoryLocalServiceImpl.java

```diff
@@ -747,6 +750,29 @@ protected void validate(
 					"There is another category named ", name,
 					" as a child of category ", parentCategoryId));
 		}
+
+		if (vocabularyId != AssetVocabularyConstants.EMPTY_VOCABULARY_ID) {
+			AssetVocabulary assetVocabulary =
+				_assetVocabularyPersistence.findByPrimaryKey(vocabularyId);
+
+			if (assetVocabulary.getGroupId() != groupId) {
+				throw new NoSuchVocabularyException(
+					StringBundler.concat(
+						"Vocabulary ", vocabularyId,
+						" does not exist in group ", groupId));
+			}
+		}
+
+		if (parentCategoryId > 0) {
+			AssetCategory parentAssetCategory = getCategory(parentCategoryId);
+
+			if (parentAssetCategory.getGroupId() != groupId) {
+				throw new NoSuchCategoryException(
+					StringBundler.concat(
+						"Category ", parentAssetCategory.getCategoryId(),
```

Comment:
(optional) Just for consistence with the previous IF block I would use `parentCategoryId` inside `concat`, or use `assetVocabulary.getVocabularyId()` in previous IF block.